### PR TITLE
azure agent support added for container apps

### PR DESCRIPTION
**Pull request by Angith**
This PR consists of the changes for supporting Azure Container Apps. azure agent will validate if the app is deployed in container Apps and send the span data to the serverless endpoint of the backend.

Introduced new 2 environment variables. `AZURE_SUBSCRIPTION_ID` and `AZURE_RESOURCE_GROUP`. These environment variables are essential for constructing your container apps' resource ID. The user needs to set these in Container Apps while deploying the image.

### CODE REVIEW

**Review by nithinputhenveettil — COMMENTED** [review]
A few minor clarifications are needed, everything else looks good. Thank you for the refactoring effort.

File: sensor.go [hunk 1]
```diff
@@ -261,25 +283,34 @@ func ShutdownSensor() {
...
+
+	// Knative, e.g. Google Cloud Run
+	case os.Getenv(kService) != "" && os.Getenv(kConfiguration) != "" &&
+		os.Getenv(kRevision) != "":
 		return newGCRAgent(serviceName, agentEndpoint, agentKey, client, logger)
-	case os.Getenv("FUNCTIONS_WORKER_RUNTIME") == azureCustomRuntime:
+
+	// azure functions or container apps
+	case os.Getenv(azureFunctionsRuntime) == azureCustomRuntime ||
+		os.Getenv(azureExecutionEnv) == azureContainerApps:
```
[the remark this answers is not in the document]
Comment (Angith) [reply sensor.go]: Added 👍

File: sensor.go [hunk 2]
```diff
@@ -28,6 +28,28 @@ const (
...
+// knative constants
+const (
+	kService       = "K_SERVICE"
+	kConfiguration = "K_CONFIGURATION"
+	kRevision      = "K_REVISION"
+)
+
+// azure constants
+const (
+	azureExecutionEnv     = "AZURE_EXECUTION_ENV"
```
[the remark this answers is not in the document]
Comment (Angith) [reply sensor.go]: Yes. This `env` is removed. However, the user needs to set 2 other environment variables in Container Apps. A readme is added for that 👍

File: docs/azure_container_apps.md [hunk 3]
```diff
@@ -0,0 +1,10 @@
+## How to Monitor Serverless Go Application Deployed in Container Apps
+
+To monitor a Go application deployed in Azure Container Apps, you can instrument the application with the Instana Go Tracer SDK and deploy it to the container apps similar to other platforms. Additionally, ensure the following environment variables are set in Container Apps to achieve proper infrastructure correlation:
```
Comment (nithinputhenveettil) [inline docs/azure_container_apps.md:3]:
```suggestion
Monitoring a Go application deployed in Azure Container Apps is straightforward and almost identical to monitoring any other application. Simply instrument your application with the Instana Go Tracer SDK, deploy it to the container apps, and ensure the necessary environment variables for infrastructure correlation are set. 
```

File: azure_container_apps_test.go [hunk 4]
```diff
@@ -0,0 +1,132 @@
+// (c) Copyright IBM Corp. 2022
```
Comment (nithinputhenveettil) [inline azure_container_apps_test.go:1]:
```suggestion
// (c) Copyright IBM Corp. 2024
```

File: docs/azure_container_apps.md [hunk 5]
```diff
@@ -0,0 +1,10 @@
+## How to Monitor Serverless Go Application Deployed in Container Apps
+
+Monitoring a Go application deployed in Azure Container Apps is straightforward and almost identical to monitoring any other application. Simply instrument your application with the Instana Go Tracer SDK, deploy it to the container apps, and ensure the necessary environment variables for infrastructure correlation are set.
```
Comment (sanojsubran) [inline docs/azure_container_apps.md:3]:
```suggestion
Monitoring a Go application deployed in Azure Container Apps is straightforward and similar to monitoring any other application. Simply instrument your application with Instana Go Tracer SDK, deploy it to the container apps, and ensure that the necessary environment variables for infrastructure correlation are set. 
```

**Review by sanojsubran — APPROVED** [review]
Looks good!

**Review by nithinputhenveettil — APPROVED** [review]

**Pull request merged**